pyaedt/modeler/cad/component_array.py

```diff
@@ -0,0 +1,718 @@
+
+
+class ComponentArray(object):
+    """Manages object attributes for a 3D component array.
+
+    Parameters
+    ----------
+    app : :class:`pyaedt.Hfss`
+        HFSS PyAEDT object.
+    name : str, optional
+        Array name. The default is ``None``, in which case a random name is assigned.
+
+    Examples
+    --------
+    Basic usage demonstrated with an HFSS design with an existing array:
+
+    >>> from pyaedt import Hfss
+    >>> aedtapp = Hfss(projectname="Array.aedt")
+    >>> array_names = aedtapp.component_array_names[0]
+    >>> array = aedtapp.component_array[array_names[0]]
+    """
+
+    def __init__(self, app, name=None):
+        # Public attributes
+        self.logger = app.logger
+        self.update_cells = True
+
+        # Private attributes
+        self.__app = app
+        if name is None:
+            name = _uname("Array_")
+        self.__name = name
+
+        # Leverage csv file if possible (aedt version > 2023.2)
+        if self.__app.settings.aedt_version > "2023.2":  # pragma: no cover
+            self.export_array_info(array_path=None)
+            self.__array_info_path = os.path.join(self.__app.toolkit_directory, "array_info.csv")
+        else:
+            self.__app.save_project()
+
+        # Data that cannot be obtained from CSV
+        try:
+            self.__cs_id = app.design_properties["ArrayDefinition"]["ArrayObject"]["ReferenceCSID"]
+        except AttributeError:  # pragma: no cover
+            self.__cs_id = 1
+
+        self.__omodel = self.__app.get_oo_object(self.__app.odesign, "Model")
+        self.__oarray = self.__app.get_oo_object(self.__omodel, name)
+        self.__array_info_path = None
+        self.__cells = None
+        self.__post_processing_cells = {}
+
+    @pyaedt_function_handler()
+    def __getitem__(self, key):
+        """Get cell object corresponding to a key (row, column).
+
+        Parameters
+        ----------
+        key : tuple(int,int)
+            Row and column associated to the cell.
+
+        Returns
+        -------
+        :class:`pyaedt.modeler.cad.component_array.CellArray`
+        """
+
+        if key[0] > self.a_size or key[1] > self.b_size:
+            self.logger.error("Specified cell does not exist.")
+            return False
+
+        if key[0] <= 0 or key[1] <= 0:
+            self.logger.error("Row and column index start with ``1``.")
+            return False
+
+        return self.cells[key[0] - 1][key[1] - 1]
+
+    @property
+    def component_names(self):
+        """List of component names."""
+        return self.properties["component"]
+
+    @property
+    def cells(self):
+        """List of :class:`pyaedt.modeler.cad.component_array.CellArray` objects."""
+        if not self.update_cells:
+            return self.__cells
+
+        if self.__app.settings.aedt_version > "2023.2":  # pragma: no cover
+            self.__array_info_path = self.export_array_info(array_path=None)
```

Comment:
Why do you update the array info path ? It seems to me that the path used is always the same.

Reply:
I prefer to update the path just in case the export array info method does not work as expected and the new CSV file is exported in any other location (for example for any permission restriction), which should not happen, but it does not affect the code. As you know, we are trusting the CSV file for loading the array information until an API method is developed for future versions.